src/super_gradients/training/models/conversion.py

```diff
@@ -48,6 +56,109 @@ def forward(self, x):
         return self.post_process(self.model(self.pre_process(x)))
 
 
+@resolve_param("pre_process", TransformsFactory())
+@resolve_param("post_process", TransformsFactory())
+def convert_to_coreml(
+    model: torch.nn.Module,
+    out_path: str,
+    input_shape: tuple = None,
+    pre_process: torch.nn.Module = None,
+    post_process: torch.nn.Module = None,
+    prep_model_for_conversion_kwargs=None,
+    export_as_ml_program=False,
+    torch_trace_kwargs=None,
+):
+    """
+        Exports a given SG model to CoreML mlprogram or package.
+
+        :param model: torch.nn.Module, model to export to ONNX.
+        :param out_path: str, destination path for the .onnx file.
+        :param input_shape: Input shape without batch dimensions ([C,H,W]). Batch size assumed to be 1.
+        DEPRECATED USE input_size KWARG IN prep_model_for_conversion_kwargs INSTEAD.
+        :param pre_process: torch.nn.Module, preprocessing pipeline, will be resolved by TransformsFactory()
+        :param post_process: torch.nn.Module, postprocessing pipeline, will be resolved by TransformsFactory()
+        :param prep_model_for_conversion_kwargs: dict, for SgModules- args to be passed to model.prep_model_for_conversion
+         prior to torch.onnx.export call. Supported keys are:
+        - input_size - Shape of inputs with batch dimension, [C,H,W] for image inputs.
+         When true, the simplified model will be saved in out_path (default=True).
+        :param export_as_ml_program: Whether to convert to the new program format (better) or legacy coreml proto file
+                            (Supports more iOS versions and devices, but this format will be deprecated at some point).
+        :param torch_trace_kwargs: kwargs for torch.jit.trace
+    :return: Path
+    """
+    if ct is None:
+        raise ImportError(
+            '"coremltools" is required for CoreML export, but is not installed. Please install CoreML Tools using:\n'
+            '   "python3 -m pip install coremltools" and try again (Tested with version 6.3.0);'
```

Comment:
Is there a reason we a are using -m syntax  and explicit python3 here? If not, I suggest to follow default installation from the official repo https://github.com/apple/coremltools#version-6

Reply:
python3 -m pip is the perefferable way to use pip, in general, not just here. 
When executin pip/pip3 you don't control the interpreter being used. 
In python, it is always better to be explicit.
You might have more than 2 python 3.X interpretrs installed.
Python3 -m pip let's OS determine which one to use.

'pip' as binary or command is implicit, might use other interpreters and venvs, and so on.
Long story short - use python3 -m pip whenver possible to avoid environment installation bugs.

Plus, I believe users understand that's replecable with pip install.